notify.js

```diff
@@ -139,16 +153,30 @@
             'requireInteraction': this.options.requireInteraction,
             'silent': this.options.silent,
             'closeOnClick': this.options.closeOnClick
-        });
+        };
 
-        if (!this.options.requireInteraction && this.options.timeout && !isNaN(this.options.timeout)) {
-            setTimeout(this.close.bind(this), this.options.timeout * 1000);
-        }
+        if (this.options.serviceWorker) {
+            var self = this;
+
+            this.options.serviceWorker.ready.then(function(registration) {
+                registration.showNotification(title, options)
+                .then(function() {
+                    return registration.getNotifications();
+                })
+                .then(function(notifications) {
+                    self.myNotify = notifications[notifications.length - 1]; // TODO: Naive implementation, improve
+                    bindNotificationEvents(self);
```

Comment:
You might be getting a bit mixed up maybe, it seems like the docs say this event should be on the SW object itself. The first paragraph says:

"The parameter passed into the onnotificationclick handler, the NotificationEvent interface represents a notification click event that is dispatched on the ServiceWorkerGlobalScope of a ServiceWorker."